### PR TITLE
English Search Result Lemma Freqency Ranking

### PR DESCRIPTION
**Pull request by Madoshakalaka**
This uses the first and second column of `W_aggr_corp_morph_log_freq.txt` and created a simple version of lemma frequency ranking for English results.

I have yet to investigate how to properly parse and use the third colum.
```
11.5713	paminêw	pam-/-n/-1
11.5728	têpihtam	têp-/-ht
11.5733	kîwêham	kîwê-/-ah
11.5743	mitâs	{mi-}/-tâs-
11.5743	mitâs	{mi-}/-tâs-
```

For each entry matched by English to be shown to the user, the frequency of the normatized lemma form will be compared.

# before
Screenshot from 2020-03-12 13-34-01

# after
Screenshot from 2020-03-12 13-34-07

### CODE REVIEW

File: .idea/webServers.xml [hunk 1]
```diff
@@ -10,6 +10,14 @@
...
         </fileTransfer>
       </webServer>
+      <webServer id="895c8cd5-e19d-4d00-8aa1-11a32e9e201d" name="ohaton" url="http://ohaton.cs.ualberta.ca">
+        <fileTransfer host="ohaton.cs.ualberta.ca" port="22" accessType="SFTP">
+          <advancedOptions>
+            <advancedOptions dataProtectionLevel="Private" passiveMode="true" shareSSLContext="true" />
+          </advancedOptions>
+          <option name="port" value="22" />
+        </fileTransfer>
+      </webServer>
```
Comment (Madoshakalaka) [inline .idea/webServers.xml]: just cuz I used ohaton to access sapir

File: CreeDictionary/API/apps.py [hunk 2]
```diff
@@ -1,9 +1,67 @@
...
+
+    lines = (
+        Path(shared_res_dir / "W_aggr_corp_morph_log_freq.txt").read_text().splitlines()
+    )
+    for line in lines:
+        cells = line.split()
+        # todo: use the third row
+        if len(cells) >= 2:
+            freq, morpheme, *_ = cells
+            Wordform.MORPHEME_RANKINGS[morpheme] = float(freq)
```
Comment (eddieantonio) [inline CreeDictionary/API/apps.py]: 👍

File: CreeDictionary/API/models.py [hunk 3]
```diff
@@ -741,7 +744,20 @@ def sort_search_result(
...
+            return 1
+        elif not a_in_rankings and b_in_rankings:
+            return -1
+        elif not a_in_rankings and not b_in_rankings:
+            return 0
+        else:  # both in rankings
+            return (
+                Wordform.MORPHEME_RANKINGS[res_a.matched_cree]
+                - Wordform.MORPHEME_RANKINGS[res_b.matched_cree]
+            )
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py]: 👍

File: CreeDictionary/tests/API_tests/model_test.py [hunk 4]
```diff
@@ -236,3 +236,11 @@ def test_search_text_with_ambiguous_word_classes():
...
     assert len(results) == 2
     assert {r.lemma_wordform.pos for r in results} == {"N", "V"}
+
+
+@pytest.mark.django_db
+def test_lemma_ranking_most_frequent_word():
+    # the English sleep should many cree words. But nipâw should show first because
+    # it undoubtedly has the highest frequency
+    results = Wordform.search("sleep")
+    assert results[0].matched_cree == "nipâw"
```
Comment (eddieantonio) [inline CreeDictionary/tests/API_tests/model_test.py]: Fair enough; we use this in demos with Antti!

**Pull request merged**